cosmic_ray/operators/unary_operator_replacement.py

```diff
@@ -27,21 +27,51 @@ def _to_ops(from_op):
             yield to_op
 
 
-class MutateUnaryOperator(Operator):
-    """An operator that modifies unary operators."""
+class ReplaceUnaryOperator(ReplacementOperator):  # pylint: disable=abstract-method
+    """An operator that modifies unary operators.
+
+    This is subclassed for each valid from/to unary operator pair, and each
+    subclass provides `from_op` and `to_op` class-level attributes. Each of
+    these is an `ast.Node` or `None`.
+    """
 
-    def visit_UnaryOp(self, node):  # pylint: disable=invalid-name
-        """
-            http://greentreesnakes.readthedocs.io/en/latest/nodes.html#UnaryOp
-        """
-        return self.visit_mutation_site(
-            node,
-            len(build_mutations([type(node.op)], _to_ops)))
+    def visit_UnaryOp(self, node):  # pylint: disable=invalid-name, missing-docstring
+        if isinstance(node.op, self.from_op):
+            return self.visit_mutation_site(node)
+        return node
 
-    def mutate(self, node, idx):
+    def mutate(self, node, _):
         "Perform the `idx`th mutation on node."
-        _, to_op = build_mutations([type(node.op)], _to_ops)[idx]
-        if to_op:
-            node.op = to_op()
-            return node
-        return node.operand
+        if self.to_op is None:
+            return node.operand
+
+        node.op = self.to_op()
+        return node
+
+
+def _create_replace_unary_operator(from_op, to_op):
+    """Create a ReplaceUnaryOperator subclasses that mutates `from_op` to `to_op`.
+    """
+    return type(
+        'ReplaceUnaryOperator_{}_{}'.format(
+            from_op.__name__,
+            'None' if to_op is None else to_op.__name__),
+        (ReplaceUnaryOperator,),
+        {'from_op': property(lambda self: from_op),
+         'to_op': property(lambda self: to_op)})
+
+
+# Create the mutation operator classes
+_MUTATION_OPERATORS = tuple(
+    _create_replace_unary_operator(_AST_OPERATORS[idx], to_op)
+    for idx, to_op in build_mutations(_AST_OPERATORS, _to_ops))
+
+
+# Add the mutation operator classes to the module namespace.
+for op in _MUTATION_OPERATORS:
```

Comment:
Consider:

    globals().update(op.__name__, op for op in _MUTATION_OPERATORS)